### PR TITLE
Metrics - Enabled encrypted OTel Endpoint

### PR DESCRIPTION
**Pull request by deepchand**
closes:#32164 
Previously, the only supported endpoint for connecting to the OpenTelemetry collector was insecure (HTTP). However, users now have the option to use a secure endpoint (HTTPS) or the insecure endpoint (HTTP)

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/metrics/otel_logger.py [hunk 1]
```diff
@@ -394,8 +395,11 @@ def get_otel_logger(cls) -> SafeOtelLogger:
     allow_list_validator = AllowListValidator(allow_list)
 
     resource = Resource(attributes={SERVICE_NAME: "Airflow"})
-    # TODO:  figure out https instead of http ??
-    endpoint = f"http://{host}:{port}/v1/metrics"
+
+    if ssl_active:
+        endpoint = f"https://{host}:{port}/v1/metrics"
+    else:
+        endpoint = f"http://{host}:{port}/v1/metrics"
```
Comment (ferruzzi) [inline airflow/metrics/otel_logger.py]: I'd either use 

```suggestion
    endpoint = f"{'https' if ssl_active else 'http'}://{host}:{port}/v1/metrics"
```


or

```suggestion
    protocol = "https" if ssl_active else "http"
    endpoint = f"{protocol}://{host}:{port}/v1/metrics"
```

**Comment by ferruzzi** [comment]
NICE!   It looks like it worked out of the box now?

@howardyoo - Check this out :)

**Comment by howardyoo** [comment]
yeah, and this is very cool :-).
True spirit of open source!

Howard

File: airflow/metrics/otel_logger.py (hunk 1 above)
Comment (deepchand) [reply airflow/metrics/otel_logger.py]: thanks @ferruzzi, I have updated the code

**Comment by deepchand** [comment]
> NICE! It looks like it worked out of the box now?
> 
> @howardyoo - Check this out :)

Yes it works out of the box at client side but at server side where otel collector is running we need to configure ssl certificate

**Comment by ferruzzi** [comment]
Is it something you might be able to demonstrate in `https://github.com/apache/airflow/blob/main/scripts/ci/docker-compose/otel-collector-config.yml` perhaps?  Or maybe explain the setup process for the user in https://github.com/apache/airflow/blob/main/docs/apache-airflow/administration-and-deployment/logging-monitoring/metrics.rst ?

**Comment by deepchand** [comment]
> Is it something you might be able to demonstrate in the Breeze integration [[here](https://github.com/apache/airflow/blob/main/scripts/ci/docker-compose/otel-collector-config.yml)] perhaps? Or maybe explain the setup process for the user in the metrics docs [[here](https://github.com/apache/airflow/blob/main/docs/apache-airflow/administration-and-deployment/logging-monitoring/metrics.rst)]? Or both? I think the user docs would be the more useful choice if you have to pick one.

i have updated the doc [[here](https://github.com/apache/airflow/blob/main/docs/apache-airflow/administration-and-deployment/logging-monitoring/metrics.rst)]

**Comment by ferruzzi** [comment]
Outstanding.   I'll approved as soon as I see a green CI.

**Review by hussein-awala — COMMENTED** [review]
Could you add a test for this change?

**Review by o-nikolas — APPROVED** [review]

**Review by ferruzzi — APPROVED** [review]
All of my concerns have been addressed.

Just one concern pending; @hussein-awala asked for unit test coverage.

**Comment by ferruzzi** [comment]
For a unit test, I don't think we can really test the connection since that would require spinning up a Collector to connect to.  Maybe there is some way to mock the connection and check the protocol it is using?

Another idea might be to move [this line](https://github.com/apache/airflow/pull/32524/files#diff-bc07cf67d2d211678aaefd4d99b1e8c6e2c6e26a6987e85c5397ccc78fed47feR400) into a helper method and just test if it returns an "https" string or not?   The connection inspection sounds like a much better answer, but the string check would be significantly easier.   

What do folks think?

**Comment by hussein-awala** [comment]
IMO a simple parameterized test which sets `otel_ssl_active` to False and True, and calls the method `get_otel_logger` should be enough, where we can mock `OTLPMetricExporter` and check the provided endpoint.

**Comment by deepchand** [comment]
@hussein-awala IMO since endpoint is intermediate variable which we want to test and it is not getting return from get_otel_logger function, @ferruzzi approach make sense to me

**Comment by potiuk** [comment]
Hmm. I am also for merging as-is - unit test for that one seems a bit "over-the-top" @hussein-awala .  I am usually sceptical to accept PRs that have no corresponding tests but this one looks like would be one of those that I see as "meh... not worth it". It's unlikely to ever fall into regression :)

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
You need to resolve conflicts now - default_airflow.cfg has been removed so  you should remove it too.

**Review by vincbeck — APPROVED** [review]

**Review by eladkal — APPROVED** [review]
LGTM

I'll take @hussein-awala thumbs up on https://github.com/apache/airflow/pull/32524#issuecomment-1642644195 as OK with not adding a test

**Comment by ferruzzi** [comment]
A bit of a bumpy road, but congrats on your first merge!

**Pull request merged**